### ISSUE TITLE
Bad spacing with horinzontal legend

### ISSUE THREAD

**Issue by cfoisy-osisoft** [plotly/plotly_express#113]
As seen in the attached screenshot:
* missing space between x-axis name and legend
* extra space between first and second line of legend

![image](https://user-images.githubusercontent.com/25752437/60769428-7a14bd80-a09d-11e9-9eeb-5c5f0bd5c45e.png)

Using version 0.3.1

**Comment by nicolaskruchten** [comment]
Yes, unfortunately the underlying engine isn't able to handle all layout cases particularly well. In this case you'd need to either manually-position the legend or manually increase the bottom margin.
